platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -1042,4 +1045,24 @@ private WritablePlatformStateStore writablePlatformStateStore() {
         }
         return store;
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void calculateHash() {
+        throwIfMutable("Hashing should only be done on immutable states");
+        throwIfDestroyed("Hashing should not be done on destroyed states");
+        if (getHash() != null) {
+            return;
+        }
+        try {
+            MerkleCryptoFactory.getInstance().digestTreeAsync(this).get();
```

Comment:
This class is deprecated, is it possible to pass in `MerkleCryptography`?

Reply:
Done